### ISSUE TITLE
localrule with --google-lifesciences does not use conda environment

### ISSUE THREAD

**Issue by cademirch** [snakemake/snakemake#1442]
**Snakemake version**

`6.15.5`
**Describe the bug**

Running snakemake with `--use-conda` and `--google-lifesciences`, `localrules` are not executed in their specified conda environment. 
**Logs**

```
(snakemake) ubuntu@cade-launchpad:/mnt/snakemake_test$ snakemake --google-lifesciences --default-remote-prefix mybucket --use-conda --google-lifesciences-region us-central1 -j1
Building DAG of jobs...
Unable to retrieve additional files from git. This is not a git repository.
Using shell: /usr/bin/bash
Provided cloud nodes: 1
Job stats:
job      count    min threads    max threads
-----  -------  -------------  -------------
test         1              1              1
total        1              1              1

Select jobs to execute...

[Mon Feb 28 17:47:59 2022]
localrule test:
    output: mybucket/done.txt
    jobid: 0
    resources: tmpdir=/tmp

/usr/bin/bash: samtools: command not found
[Mon Feb 28 17:47:59 2022]
Error in rule test:
    jobid: 0
    output: mybucket/done.txt
    shell:
        samtools --help > mybucket/done.txt
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job test since they might be corrupted:
mybucket/done.txt
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: /mnt/snakemake_test/.snakemake/log/2022-02-28T174755.754811.snakemake.log
```
**Minimal example**

**Snakefile:**
```
localrules: test
rule test:
    output: "done.txt"
    conda: "env.yml"
    shell: "samtools --help > {output}"
```
**env.yml:**
```
channels:
  - bioconda
  - defaults
  - conda-forge
dependencies:
  - samtools>=1.14
```
**Additional context**

**Comment by nate-d-olson** [comment]
I believe this bug is also impacting local rules when using tibanna, #1302

**Comment by cademirch** [comment]
I looked into this a little bit today. Still don't really know what could be causing this... it seems that here:
https://github.com/snakemake/snakemake/blob/2a0230de243d4b4bf21b6216c5f172a3b4075abf/snakemake/shell.py#L129-L161
`conda_env` is None. So for some reason the rule function context does not have `conda_env` set. Will do some more digging but could use any insight!

**Comment by cademirch** [comment]
It seems that the issue is probably earlier than my above post, since Snakemake does not even create the conda env. Which when Snakemake runs successfully, happens first.

**Comment by cademirch** [comment]
I believe the problem could begin here:
https://github.com/snakemake/snakemake/blob/2a0230de243d4b4bf21b6216c5f172a3b4075abf/snakemake/workflow.py#L967-L971

`self.assume_shared_fs` is False - which makes sense because I am using the google lifesciences executor. However, that prevents us from calling `dag.create_conda_envs`. In `dag.create_conda_envs` there is a check if the job is local:
https://github.com/snakemake/snakemake/blob/2a0230de243d4b4bf21b6216c5f172a3b4075abf/snakemake/dag.py#L283
But that doesn't matter, since we never call the function.

Commenting out the check for `self.assume_shared_fs` allows the MRE above to run just fine. I'm curious why that check is there to begin with - that would be helpful to know since I'm not sure what unintended consequences there may be in removing it.

**Comment by cademirch** [comment]
@nate-d-olson Could you test this fix on Tibanna? I imagine it should work.

**Comment by cademirch** [comment]
Still an issue in 7.2.1. @johanneskoester Could you take a look at this? It seems that the check for shared_fs was implemented b/c of issue #1463. Though I'm curious if the check needs to happen since this line:
https://github.com/snakemake/snakemake/blob/2a0230de243d4b4bf21b6216c5f172a3b4075abf/snakemake/dag.py#L283
Should solve the issue of creating the same conda env many times? Not sure though.

**Comment by johanneskoester** [comment]
Thanks a lot. I agree, and have implemented the fix in #1490.
